bin/main.ml

```diff
@@ -769,10 +843,67 @@ let build_targets =
     in
     set_common common ~targets;
     let log = Log.create common in
-    Scheduler.go ~log ~common
-      (Main.setup ~log common >>= fun setup ->
+
+    let setup = Main.setup ~log common in
+
+    let build_once setup =
        let targets = resolve_targets_exn ~log common setup targets in
-       do_build setup targets)
+       do_build setup targets
+    in
+
+    let clear_all_caches () =
+      Dir_contents.clear_cache ();
+      Report_error.clear_cache ()
+    in
+
+    let build_poll =
+      let rec loop () =
+        (setup
+         >>= fun setup ->
+         build_once setup
+         >>= fun () ->
+         fswatch_command File_tree.(Dir.path (root setup.file_tree))
+         >>= fun (fswatch, args) ->
+         let fswatch =
```

Comment:
We should do this outside of the loop. `Dir.path (root setup.file_tree)` is always `Path.root`